### PR TITLE
#69 feature: Login with Email and Password

### PR DESCRIPTION
**Pull request by oliviertech212**
## What does this PR do:
- create login react component
- design login UI
- test ui component
## Descriptions
- Login with Email and Password
## How should this be manually tested?
- clone the repo by ` git clone  https://github.com/atlp-rwanda/ecommerce-app-champions-fe.git`
- change branch git checkout `ft-signin`
- create .env file and paste variables from .env.example
- run `npm run dev`
-  copy url link into your browser `url/Login`
Screenshot 2023-04-26 at 23 26 32

-  run test `npm run test`
- run coverage `npm run coverage`
Screenshot 2023-04-26 at 23 25 38

### CODE REVIEW

**Comment by oliviertech212** [comment]
[Users should be able to Login with Email and Password](https://trello.com/c/LRMtWOCV/69-users-should-be-able-to-login-with-email-and-password)

File: vite.config.js
Comment (sostene12) [inline vite.config.js:1]: please recall on the commented codes  and change that

**Review by sostene12 — COMMENTED** [review]
- login with Google is not working

File: vite.config.js
Comment (oliviertech212) [reply vite.config.js:1]: create .env file and paste variables from .env.example

**Review by chrill-christella — APPROVED** [review]

File: src/pages/Login.jsx [hunk 1]
```diff
@@ -1,9 +1,92 @@
...
+							type={field.type}
+							id={field.id}
+							name={field.name}
+							value={loginState[field.id]}
+							isRequired={field.isRequired}
+							labelText={field.labelText}
+							labelFor={field.labelFor}
+							autoComplete={field.autoComplete}
+							handleChange={handleChange}
+							className="appearance-none py-2 px-2 my-2 rounded-md w-11/12"
```
Comment (wakaflorien) [inline src/pages/Login.jsx]: - This input seems to be wider than the ones signup.

File: src/redux/reducers/Auth/loginSlice.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			state.error = null;
+		},
+		loginSuccess: (state, action) => {
+			state.loading = false;
+			state.error = null;
+			state.user = action.payload;
+		},
+		loginFail: (state, action) => {
+			state.loading = false;
+			state.user = '';
```
Comment (wakaflorien) [inline src/redux/reducers/Auth/loginSlice.js]: - use `null` instead of empty string.

File: src/pages/Login.jsx [hunk 3]
```diff
@@ -1,9 +1,92 @@
...
+		setLoginState({ ...loginState, [e.target.id]: e.target.value });
+
+	const handleLogin = (e) => {
+		e.preventDefault();
+		dispatch(login(loginState));
+	};
+
+	useEffect(() => {
+		setTimeout(() => {
+			if (user?.data?.others?.RoleId === 1) return navigate('/admin');
```
Comment (wakaflorien) [inline src/pages/Login.jsx]: - better to use `redirect` than this `useNavigate` hook.

https://reactrouter.com/en/main/fetch/redirect

File: src/redux/actions/auth.loginActions.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import Cookies from 'js-cookie';
```
Comment (wakaflorien) [inline src/redux/actions/auth.loginActions.js:1]: - address eslint errors on this line.

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- rename the branch to follow the the guide
- add `coverage` directory in `.gitignore` @oliviertech212

**Comment by oliviertech212** [comment]
[WIP - Users should be able to Login via Google](https://trello.com/c/V4aK1Xhh/68-wip-users-should-be-able-to-login-via-google)

File: vite.config.js [hunk 5]
```diff
@@ -1,3 +1,20 @@
+// import { defineConfig } from 'vite';
```
Comment (wakaflorien) [inline vite.config.js]: - remove these commented line @oliviertech212

**Review by wakaflorien — APPROVED** [review]

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- rename the branch

**Review by sostene12 — APPROVED** [review]

**Review by Shumba250 — APPROVED** [review]

**Pull request closed without merging**